icu4c/source/tools/ctestfw/tstdtmod.cpp

```diff
@@ -88,12 +111,14 @@ void IcuTestErrorCode::setScope(const UnicodeString& message) {
 }
 
 void IcuTestErrorCode::handleFailure() const {
-    errlog(FALSE, nullptr);
+    errlog(FALSE, u"", nullptr);
 }
 
-void IcuTestErrorCode::errlog(UBool dataErr, const char* extraMessage) const {
+void IcuTestErrorCode::errlog(UBool dataErr, UnicodeString mainMessage, const char* extraMessage) const {
     UnicodeString msg(testName, -1, US_INV);
-    msg.append(u" failure: ").append(UnicodeString(errorName(), -1, US_INV));
+    msg.append(u" ");
```

Comment:
Optional: You could use the UChar append and chain the string append:

msg.append(u' ').append(mainMessage);

Reply:
Done.